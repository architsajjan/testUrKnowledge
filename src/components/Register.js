// IMPORTS
import React, { Component } from 'react';
import PropTypes from "prop-types";

export default class Register extends Component {
    constructor(props){
        super(props);
        this.state={
            fullName:"",
            email:"",
            contact:"",
        };
        this.enableSubmit = false;                 
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.validateForm = this.validateForm.bind(this);
    }

    // VALIDATE INPUTS AND ENABLE SUBMIT BUTTON
    validateForm(){
        const { fullName, email, contact } = this.state;
        if( fullName.length > 3 &&
            email.length > 7 &&
            contact.length > 9
        ){
            this.enableSubmit = true;
        }
        else {
            this.enableSubmit = false;
        }
        
    }

    handleChange(event) {
        this.setState({[event.target.name]: event.target.value});
    }

    handleSubmit(){
        const {fullName, email, contact} = this.state;
        this.validateForm();

        if(this.enableSubmit)
            this.setState(
                () => {
                this.props.history.push({
                    pathname: `/test-instructions`,
                    state: {fullName, email, contact}
                });
                }
            );
        
    }

    render(){        
        return (
            <div className="landing">
                <div className="">
                    <div className="landing-main dark-overlay">
                    <h1 className="x-large">Test Ur Knowledge</h1>
                    <p className="lead">
                    Enter your Details to Continue with the test:
                    </p>
                    </div>
                </div>
                <div className="landing-side">
                    <div className="container">
                        <h1 className="large text-primary">Register</h1>
                        <form className="form" onSubmit={this.handleSubmit}>
                            <div className="form-group">
                                <input type="text" placeholder="Full Name" name="fullName" onChange={this.handleChange} spellCheck={false} required autoFocus/>
                            </div>
                            <div className="form-group">
                                <input type="email" placeholder="Email Address" name="email" onChange={this.handleChange} required/>
                            </div>
                            <div className="form-group">
                                <input type="text" placeholder="Contact No." name="contact" onChange={this.handleChange} minLength="10" max="10000000000" required/>
                            </div>
                            <div className="buttons">
                                <input type="button" className="btn btn-primary" onClick={this.handleSubmit} value="Submit Details" />
                            </div>
                        </form>
                        
                    </div>    
                    
                </div>
            </div>
        );
    }
}

Register.propTypes = {
    history: PropTypes.object,
  };