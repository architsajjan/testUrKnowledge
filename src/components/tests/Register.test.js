import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { BrowserRouter } from "react-router-dom";

import Register from '../Register';

it("runs without crash", () => {
  const div = document.createElement("div");
  render(<BrowserRouter><Register history={{}} /></BrowserRouter>, div);
  unmountComponentAtNode(div);
});