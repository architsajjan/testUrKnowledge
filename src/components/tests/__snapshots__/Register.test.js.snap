// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="landing"
>
  <div
    className=""
  >
    <div
      className="landing-main dark-overlay"
    >
      <h1
        className="x-large"
      >
        Test Ur Knowledge
      </h1>
      <p
        className="lead"
      >
        Enter your Details to Continue with the test:
      </p>
    </div>
  </div>
  <div
    className="landing-side"
  >
    <div
      className="container"
    >
      <h1
        className="large text-primary"
      >
        Register
      </h1>
      <form
        className="form"
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            autoComplete="off"
            autoFocus={true}
            name="fullName"
            onChange={[Function]}
            placeholder="Full Name"
            required={true}
            spellCheck={false}
            type="text"
          />
          <p
            className="errPara"
          >
            
          </p>
        </div>
        <div
          className="form-group"
        >
          <input
            autoComplete="off"
            name="email"
            onChange={[Function]}
            placeholder="Email Address"
            required={true}
            type="email"
          />
          <p
            className="errPara"
          >
            
          </p>
        </div>
        <div
          className="form-group"
        >
          <input
            autoComplete="off"
            max="10000000000"
            minLength="10"
            name="contact"
            onChange={[Function]}
            placeholder="Contact No."
            required={true}
            type="text"
          />
          <p
            className="errPara"
          >
            
          </p>
        </div>
        <div
          className="buttons"
        >
          <input
            className="btn btn-primary"
            onClick={[Function]}
            type="button"
            value="Submit Details"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
